lookup/fileenv.go

```diff
@@ -82,23 +79,49 @@ func parseCustomEnvFile(file string, parent config.EnvironmentLookup) (*FileEnvL
 		return nil, scanner.Err()
 	}
 
-	return &FileEnvLookup{
-		parent:    parent,
-		variables: variables,
-	}, nil
+	return variables, nil
 }
 
-func NewFileEnvLookup(file string, parent config.EnvironmentLookup) (*FileEnvLookup, error) {
+func ParseEnvFile(file string) (map[string]interface{}, error) {
 	if file != "" {
 		if strings.HasSuffix(file, ".yml") || strings.HasSuffix(file, ".yaml") || strings.HasSuffix(file, ".json") {
-			return parseMultiLineEnv(file, parent)
+
+			v, err := ParseMultiLineEnv(file)
+			if err != nil {
+				return nil, err
+			}
+
+			return v, nil
+		}
+
+		v, err := ParseCustomEnvFile(file)
+		if err != nil {
+			return nil, err
+		}
+
+		return v, nil
+	}
+	return map[string]interface{}{}, nil
+}
+
+func NewFileEnvLookup(file string, parent config.EnvironmentLookup) (*FileEnvLookup, error) {
+
+	v, err := ParseEnvFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	variables := map[string]string{}
```

Comment:
Couldn't `ParseEnvFile` just return `map[string]string`?

Reply:
ParseEnvFile is used elsewhere that requires type `map[string]interface{}`